crates/dojo-world/src/migration/mod.rs

```diff
@@ -35,6 +33,16 @@ pub struct DeployOutput {
     pub block_number: Option<u64>,
     pub contract_address: FieldElement,
     pub declare: Option<DeclareOutput>,
+    // base class hash at time of deployment
+    pub base_class_hash: FieldElement,
+}
+
+#[derive(Clone, Debug)]
+pub struct UpgradeOutput {
```

Comment:
We should see how to use that on the dojo contracts. Currently even if upgraded, we return a `DeployOutput` struct only.